static/src/javascripts-legacy/projects/commercial/modules/dfp/prepare-switch-tag.js

```diff
@@ -74,19 +74,25 @@ define([
     function pushAdUnit(dfpDivId, sizeMapping) {
 
         var __switch_zero = window.__switch_zero;
+        var promises = [];
 
         if (__switch_zero) {
             var adUnitIds = findAdUnitIds(sizeMapping.size);
 
             adUnitIds.forEach(function(adUnitId) {
                 if (adUnitId){
-                    __switch_zero.units.push({
-                        dfpDivId: dfpDivId,
-                        switchAdUnitId: adUnitId
-                    });
+                    promises.push(new Promise(function(resolve){
+                        __switch_zero.units.push({
```

Comment:
I wonder if an exception will be swallowed here, should it happen? I think inside a resolver function, you have to explicitly call `reject` so that the error can be caught later on.